### PR TITLE
Fix coverage files combining

### PR DESCRIPTION
**Pull request by saschagrunert**
Hey, this enhancement should fix https://github.com/onsi/ginkgo/issues/518.

### CODE REVIEW

**Comment by williammartin** [comment]
Thanks for the PR. Is there a test we can write to get coverage of this?

**Comment by saschagrunert** [comment]
Yeah sure, I will add an integration test.

**Comment by saschagrunert** [comment]
@williammartin I am wondering why the test on go 1.9 fails. Could it be possible that it is the case because the travis machine still installs from `github.com/onsi/ginkgo/ginkgo` and runs this executable?

I guess for integration testing of forks the binary needs to be built locally, right?

**Comment by NilsJPWerner** [comment]
Any updates on when this will be fixed/merged?

**Comment by williammartin** [comment]
I can reproduce this failure on go 1.10.7 by focussing the changed spec. Will keep digging.

**Comment by saschagrunert** [comment]
@williammartin Is there anything I can do here? I guess the failing pipeline is related to the go fetch source...

**Comment by williammartin** [comment]
Well, I was just having a look at the contents of this file: https://github.com/onsi/ginkgo/pull/552/files#diff-28605e74bddc91d19789edefa99640b8R121

And indeed, in 1.10 it looks like this:

```
mode: atomic
mode: atomic
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/first_package/coverage.go:3.17,5.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/first_package/coverage.go:7.17,9.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/first_package/coverage.go:11.17,13.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/first_package/coverage.go:15.17,17.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/first_package/coverage.go:19.17,21.2 1 0
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/second_package/coverage.go:3.17,5.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/second_package/coverage.go:7.17,9.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/second_package/coverage.go:11.17,13.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/second_package/coverage.go:15.17,17.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/second_package/coverage.go:19.17,21.2 1 1
```

and in 1.11:

```
mode: atomic
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/first_package/coverage.go:3.17,5.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/first_package/coverage.go:7.17,9.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/first_package/coverage.go:11.17,13.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/first_package/coverage.go:15.17,17.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/first_package/coverage.go:19.17,21.2 1 0
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/second_package/coverage.go:3.17,5.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/second_package/coverage.go:7.17,9.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/second_package/coverage.go:11.17,13.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/second_package/coverage.go:15.17,17.2 1 1
github.com/onsi/ginkgo/integration/_fixtures/combined_coverage_fixture/second_package/coverage.go:19.17,21.2 1 1
```

Something funny in 1.10 is resulting in keeping the duplicate `mode: atomic` lines

**Comment by williammartin** [comment]
For some reason the coverage file in Go 1.10 has `mode: atomic` in it already.

https://github.com/onsi/ginkgo/blob/master/ginkgo/run_command.go#L166

**Comment by saschagrunert** [comment]
I must say this makes me not happy. So do we have to add a hack-around for go 1.10/11?

**Comment by williammartin** [comment]
I don't fully understand what is causing the difference yet. Ideally I'd like to reproduce a simple case of this behaviour without ginkgo to ensure I understand where the difference is and what the behaviour is.

**Comment by rohitsakala** [comment]
Hi, any update on this issue ?

**Comment by zhandao** [comment]
Looking forward to merging it

**Comment by williammartin** [comment]
I do not have the time to look at this issue right now (sorry for the lack of communication), if we want to merge this soon, I'll need help debugging the failing builds.

Thanks!

File: ginkgo/run_command.go [hunk 1]
```diff
@@ -163,17 +164,27 @@ func (r *SpecRunner) combineCoverprofiles(runners []*testrunner.TestRunner) erro
 
 	fmt.Println("path is " + path)
 
-	combined, err := os.OpenFile(filepath.Join(path, r.getCoverprofile()),
-		os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0666)
+	combined, err := os.OpenFile(
+		filepath.Join(path, r.getCoverprofile()),
+		os.O_WRONLY|os.O_CREATE,
```
Comment (saschagrunert) [inline ginkgo/run_command.go]: Fixed the go 1.10 bug by removing `os.O_APPEND`

**Comment by saschagrunert** [comment]
Rebased on top of the latest master and applied a fix for the go1.10 issue.

**Comment by williammartin** [comment]
LGTM thanks

**Pull request merged**